### PR TITLE
[feat]: [CCM-7875]: Add CCM related resources, roles and permissions

### PR DESCRIPTION
**Pull request by svmav**
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>
  
- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>
  
  You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`
  
- Compile: `trigger compile`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
</details>

---
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/34050)

### CODE REVIEW

File: access-control/service/src/main/resources/io/harness/accesscontrol/permissions/permissions.yml [hunk 1]
```diff
@@ -984,4 +984,124 @@ permissions:
...
+    includedInAllRoles: false
+    allowedScopeLevels:
+      - account
+  - identifier: ccm_budget_create
+    name: Create a CCM Budget
+    status: ACTIVE
+    includedInAllRoles: false
+    allowedScopeLevels:
+      - account
+  - identifier: ccm_budget_edit
```
Comment (hitesharinga) [inline access-control/service/src/main/resources/io/harness/accesscontrol/permissions/permissions.yml:1024]: budget permissions are present twice.

**Review by hitesharinga — CHANGES_REQUESTED** [review]

File: access-control/service/src/main/resources/io/harness/accesscontrol/permissions/permissions.yml (hunk 1 above)
Comment (svmav) [reply access-control/service/src/main/resources/io/harness/accesscontrol/permissions/permissions.yml:1012]: removed duplicate permissions

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Passed_** 
 


            
 #### 3 Issues 
* Anti-pattern 3 Anti-patterns 

 [Details](https://app.embold.io/organization/gh/harness/repositories/7752dbb3099d2dba3b5d37f5931a5e4c/pull-request/34050)

**Comment by svmav** [comment]
trigger ti1

**Review by reetika04 — APPROVED** [review]
> Reviewed 1 of 14 files at r4, 26 of 26 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/34050)* status: 1 change requests. 1. unresolved discussions.

**Review by hitesharinga — COMMENTED** [review]
> Reviewed 1 of 14 files at r4, 26 of 26 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/34050)* status: 1 change requests. 1. unresolved discussions.

**Review by hitesharinga — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/34050)* status: :shipit: complete! ✓

**Pull request merged**